xray/usage/reportusage.go

```diff
@@ -0,0 +1,177 @@
+package usage
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	versionutil "github.com/jfrog/gofrog/version"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/http/httpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray"
+)
+
+const (
+	minXrayVersion = "3.80.0"
+
+	xrayUsageApiPath = "api/v1/usage/events/send"
+
+	EcosystemUsageBaseUrl = "https://usage-ecosystem.jfrog.info/"
```

Comment:
do we actually need them to be in separate vars?

Reply:
merged